presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/SemiJoinMatcher.java

```diff
@@ -66,6 +74,7 @@ public String toString()
                 .add("filteringSymbolAlias", filteringSymbolAlias)
                 .add("sourceSymbolAlias", sourceSymbolAlias)
                 .add("outputAlias", outputAlias)
+                .add("distributionType", distributionType.orElse(null))
```

Comment:
Just `distributionType`. It is somehow confusing to put `null` here. It would imply that the `distributionType` field is set to `null`, but not to `Optional.empty()`.

Reply:
You can use `toStringHelper`'s omit nulls option.

Reply:
I would display the object state verbatim in the `toString`, unless there's a good reason not to do this (for example special toString formating, like for PlanNodeId, where toString is implemented as just an `id` string). 

This tricks with hiding fields, converting `Optional.empty` to `null`, etc. doesn't give us anything practical (maybe a little bit shorter toString() output?), but can be very confusing when debugging. 

Whenever you see some field missing in the `toString` output you need to think what happened. Whether there was bug in the code, and that field wasn't added to the `toString` output, or whether somehow that field become null (for example some older version of code could have the null check missing, or smthng).

Reply:
Also did this to be the same as JoinMatcher.  Happy to switch both.